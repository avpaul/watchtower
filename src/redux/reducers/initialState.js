import { ALL } from '../constants/fellowFilters';

export const INDEX_PAGE_SIZE_DEFAULT = 20;
export const INDEX_PAGE_SIZE_OPTIONS = ['5', '10', '19', '30', '50', '100'];

const genericAPIStateArray = {
  loading: false,
  data: [],
  error: null
};

const genericAPIStateObject = {
  loading: false,
  data: {},
  error: null
};

export default {
  emsDashboard: {
    fellowsSummary: {
      loading: false,
      summary: [],
      error: ''
    }
  },
  fellows: {
    pagination: {
      perPage: 25,
      page: 1,
      pages: 1,
      firstPageURL: '',
      nextPageURL: '',
      prevPageURL: '',
      finalPageURL: ''
    },
    fellows: [],
    summary: { fellowsofftrack: 0, gteWk5OffTrack: 0, fellowsontrack: 0 },
    loading: false,
    error: ''
  },
  visibilityFilter: ALL,
  fellowCountHistory: {
    countSummary: {},
    loading: false
  },
  fellowsProgress: {
    loading: false,
    data: {
      D0A: [],
      D0B: [],
      D0: []
    },
    error: null
  },
  fellowDevPulse: {
    loading: false,
    ratings: [],
    averageRatings: {
      communication: 'N/A',
      integration: 'N/A',
      quantity: 'N/A',
      quality: 'N/A',
      professionalism: 'N/A',
      initiative: 'N/A'
    },
    error: null
  },
  fellowLmsSubmissions: {
    loading: false,
    lmsSubmissions: {},
    error: null
  },
  manager: {
    loading: false,
    data: {},
    error: null
  },
  opsSummary: {
    data: {
      managers: {
        ttls: [],
        lfs: [],
        averageFellowsPerTtl: 0,
        averageFellowsPerLf: 0
      },
      fellowsCount: {
        Total: 0
      },
      locations: []
    },
    loading: false
  },
  managerFellows: {
    loading: false,
    managerFellows: {},
    error: ''
  },
  fellow: {
    loading: false,
    fellow: {},
    error: ''
  },
  d1Fellow: {
    loading: false,
    fellow: {},
    error: ''
  },
  notification: {
    loading: false,
    notification: [],
    error: ''
  },
  unreadnotification: {
    loading: false,
    unreadnotification: {},
    error: ''
  },
  readnotification: {
    loading: false,
    readnotification: {},
    error: ''
  },
  fellowsSummary: {
    loading: false,
    fellowsSummaryToday: {},
    fellowsSummaryTrend: {},
    data: {},
    error: ''
  },
  engineeringManagerSimsLeads: {
    loading: false,
    data: {
      managers: {},
      locations: []
    },
    error: null
  },
  updateManagerNotification: {
    loading: false,
    managerNotification: [],
    error: null
  },
  managerFellowsSummary: {
    loading: false,
    data: [],
    error: null
  },
  feedback: {
    loading: false,
    data: [],
    error: null
  },
  fellowFeedback: {
    feedback: [],
    fellowFeedback: {}
  },
  pipData: {
    data: {},
    loading: false,
    error: null
  },
  performanceData: {
    loading: false,
    data: {
      today: {},
      trend: {}
    },
    error: ''
  },
  fellowPipFeedback: {
    loading: false,
    feedback: [],
    error: null
  },
  fellowPrePipFeedback: {
    loading: false,
    feedback: [],
    error: null
  },
  pipDeactivation: {
    success: null
  },
  createProject: genericAPIStateObject,
  createProjectVacancies: genericAPIStateObject,
  allProjects: genericAPIStateArray,
  fetchProjectManagers: genericAPIStateArray,
  fetchProjectTechnologies: genericAPIStateArray,
  editSingleProject: genericAPIStateObject,
  singleProject: genericAPIStateObject,
  createCertificactionVacancies: genericAPIStateObject,
  addProjectManager: {
    manager: {}
  },
  addProjectTechnology: {
    technology: ''
  },
  fetchSlackChannels: genericAPIStateArray,
  cadreEngineers: {
    loading: false,
    cadreEngineers: [],
    error: ''
  },
  cadreVacancies: genericAPIStateObject,
  reports: {
    loading: false,
    error: '',
    data: [],
    meta: {
      page: 1,
      perPage: INDEX_PAGE_SIZE_DEFAULT,
      perPageOptions: INDEX_PAGE_SIZE_OPTIONS,
      total: 0
    }
  },
  allRoles: genericAPIStateArray,
  fetchAllRoles: genericAPIStateArray,
  fetchActiveRole: genericAPIStateArray,
  createRole: genericAPIStateArray,
  roleSkills: genericAPIStateArray,
  getAllVacancies: genericAPIStateObject,
  allCetifications: genericAPIStateArray,
  editCertification: genericAPIStateObject,
  certifications: genericAPIStateArray,
  getCertification: genericAPIStateObject,
  editProjectVacancies: genericAPIStateObject,
  deleteProjectVacancies: genericAPIStateObject,
  projectVacanciesOnFocus: {},
  deleteCertification: genericAPIStateObject,
  certificationOnFocus: {},
  certificationApplicants: genericAPIStateArray,
  certificationApplication: genericAPIStateObject
};
