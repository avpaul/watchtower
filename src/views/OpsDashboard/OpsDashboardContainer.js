import { connect } from 'react-redux';
import OpsDashboardMain from './OpsDashboard';
import getManagers from '../../redux/actionCreators/managerActions';

export const mapStateToProps = ({ managers }) => ({
  loading: managers.loading,
  ttls: managers.ttls,
  lfs: managers.lfs,
  averageFellowsPerTtl: managers.averageFellowsPerTtl,
  averageFellowsPerLf: managers.averageFellowsPerLf
});

export default connect(
  mapStateToProps,
  { getManagers }
)(OpsDashboardMain);
