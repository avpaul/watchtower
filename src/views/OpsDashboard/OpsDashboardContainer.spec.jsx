import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import initialState from '../../redux/reducers/initialState';
import OpsDashboardMain from '.';
import OpsDashboardContainer, {
  mapStateToProps
} from './OpsDashboardContainer';

const initState = initialState;
const mockStore = configureStore();
let store;

describe('OpsDashboardContainer component', () => {
  let wrapper;

  beforeEach(() => {
    store = mockStore(initState);
    wrapper = shallow(<OpsDashboardContainer store={store} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map state to props', () => {
    const tree = mapStateToProps(initialState);
    expect(tree).toMatchSnapshot();
  });

  it('renders OpsDashboardMain component', () => {
    expect(OpsDashboardMain).toBeDefined();
  });

  it('should hold initial state for component', () => {
    expect(wrapper.props().ttls).toEqual([]);
    expect(wrapper.props().lfs).toEqual([]);
    expect(wrapper.props().averageFellowsPerTtl).toEqual(0);
    expect(wrapper.props().averageFellowsPerLf).toEqual(0);
  });
});
