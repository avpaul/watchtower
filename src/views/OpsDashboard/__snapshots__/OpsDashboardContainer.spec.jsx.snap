// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpsDashboardContainer component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(OpsDashboardMain)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "averageFellowsPerLf": 0,
      "averageFellowsPerTtl": 0,
      "getManagers": [Function],
      "lfs": Array [],
      "loading": false,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "ttls": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "averageFellowsPerLf": 0,
        "averageFellowsPerTtl": 0,
        "getManagers": [Function],
        "lfs": Array [],
        "loading": false,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "ttls": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`OpsDashboardContainer component should map state to props 1`] = `
Object {
  "averageFellowsPerLf": 0,
  "averageFellowsPerTtl": 0,
  "lfs": Array [],
  "loading": false,
  "ttls": Array [],
}
`;
